script.speedtester/resources/lib/addon.py

```diff
@@ -0,0 +1,921 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at: http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+# Original speedtest-cli DEV by Matt Martz (https://github.com/sivel/speedtest-cli)
+
+from __future__ import absolute_import, division, unicode_literals
+
+import os
+import sys
+import math
+import platform
+import socket
+import threading
+import timeit
+
+try:  # Python 3
+    from xml.etree import ElementTree
+except ImportError:
+    try:  # Python 2
+        from xml.etree import cElementTree as ElementTree
+    except ImportError:
+        from xml.dom import minidom
+
+try:  # Python 3
+    from hashlib import md5
+except ImportError:  # Python 2
+    from md5 import md5
+
+try:  # Python 3
+    from queue import Queue
+except ImportError:  # Python 2
+    from Queue import Queue
+
+try:  # Python 3
+    from http.client import HTTPConnection, HTTPSConnection
+    from urllib.parse import parse_qs, urlparse
+    from urllib.request import urlopen, Request, HTTPError, URLError
+except ImportError:
+    try:  # Python 2.6 and newer
+        from httplib import HTTPConnection, HTTPSConnection
+        from urllib2 import urlopen, Request, HTTPError, URLError
+        from urlparse import parse_qs, urlparse
+    except ImportError:  # Python 2.5 and older
+        # pylint: disable=ungrouped-imports
+        from cgi import parse_qs  # pylint: disable=deprecated-method
+        from httplib import HTTPConnection, HTTPSConnection
+        from urllib.parse import urlparse
+        from urllib2 import urlopen, Request, HTTPError, URLError
+
+from xbmc import Monitor, translatePath
```

Comment:
This needs to come from xbmcvfs in matrix and above

Reply:
Hmm, I was sure we fixed that. It appears it was badly merged.